conjure-java-core/src/main/java/com/palantir/conjure/java/types/AliasGenerator.java

```diff
@@ -244,6 +255,79 @@ private static boolean isAliasOfDouble(AliasDefinition typeDef) {
                 && typeDef.getAlias().accept(TypeVisitor.PRIMITIVE).equals(PrimitiveType.DOUBLE);
     }
 
+    private static final class DefensiveCollectionConstructorVisitor implements Visitor<Optional<CodeBlock>> {
+        private final ClassName thisClass;
+        private final Options options;
+        private final CodeBlock nonNullValueExpression;
+
+        DefensiveCollectionConstructorVisitor(ClassName thisClass, Options options) {
+            this.thisClass = thisClass;
+            this.options = options;
+            this.nonNullValueExpression = Expressions.requireNonNull("value", "value cannot be null");
+        }
+
+        @Override
+        public Optional<CodeBlock> visitPrimitive(PrimitiveType _value) {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<CodeBlock> visitOptional(OptionalType _value) {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<CodeBlock> visitList(ListType value) {
+            if (options.nonNullCollections()) {
+                return Optional.of(CodeBlock.of(
+                        "return new $T($T.newNonNullList($L))",
```

Comment:
Can we specialize this taking advantage of `newNonNullIntegerList`/`newNonNullDoubleList`/`newNonNullSafeLongList` where possible? That can reduce the overhead of data on heap by avoiding boxed types.